utils/UpgradeInstruction.py

```
@@ -153,27 +174,66 @@ def generate_upgrade_instruction(base_package: str, target_version: str) -> dict
     for dep in requires_dist:
         try:
             req = Requirement(dep)
-            ver = _extract_min_version(req)
-            if ver:
-                deps.append(f"{req.name}=={ver}")
-        except Exception as e:  # pragma: no cover - lenient parse
-            logger.warning(f"Failed to parse dependency {dep}: {e}")
+            pkg_name = req.name.lower()
+            
+            # Check if we should skip this dependency
+            current_version = current_versions.get(pkg_name)
```

Ruff (0.11.9)

Undefined name `current_versions`

(F821)